src/test/java/org/wso2/securevault/secret/SecretManagerTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.securevault.secret;
+
+import org.powermock.reflect.Whitebox;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.securevault.SecureVaultException;
+
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Unit test class for SecretManager.
+ */
+public class SecretManagerTest {
+
+    private SecretManager secretManager;
+
+    @BeforeClass
+    public void setUp() {
+
+        secretManager = SecretManager.getInstance();
+    }
+
+    @Test(description = "Test case for filterConfigurations() method.")
+    public void testFilterConfigurations() throws Exception {
+
+        Properties actual = Whitebox.invokeMethod(secretManager, "filterConfigurations", "secretProviders",
+                getConfigProperties());
+        Assert.assertEquals(12, actual.size());
+    }
+
+    @Test(description = "Test case for populateArray() method.")
+    public void testPopulateArray() throws Exception {
+
+        String[] actual = Whitebox.invokeMethod(secretManager, "populateArrayOfSecretProviders", "vault,hsm");
+        Assert.assertEquals(2, actual.length);
+    }
+
+    @Test(description = "Test case for resolveSecret() method.")
+    public void testResolveSecretNovel() throws Exception {
+
+        Whitebox.invokeMethod(secretManager, "readLegacyProviders", getConfigProperties());
+        Whitebox.invokeMethod(secretManager, "readNovelProviders", getConfigProperties());
+        String alias = secretManager.resolveSecret("vault:hashicorp:admin_password");
+        Assert.assertEquals("admin_password", alias);
+    }
+
+    @Test(description = "Test case for resolveSecret() method.")
+    public void testResolveSecretLegacy() throws Exception {
+
+        Whitebox.invokeMethod(secretManager, "readLegacyProviders", getConfigProperties());
+        Whitebox.invokeMethod(secretManager, "readNovelProviders", getConfigProperties());
+        String alias = secretManager.resolveSecret("admin_password");
+        Assert.assertEquals("admin_password", alias);
+    }
+
+    @Test(description = "Negative test case for resolveSecret() method.")
```

Comment:
Isn't expecting exception is not the correct way ?

Reply:
Changed